### ISSUE TITLE
VsVim for Mac: relativenumber option is not working

### ISSUE THREAD

**Issue by funseiki** [VsVim/VsVim#2771]
**Describe the bug**
`set relativenumber` does not appear to have any effect on the editor (line numbers are shown with absolute ordering).

**To Reproduce**

Via vimrc
1. Add `set relativenumber` to ~/.vimrc (I also added it to ~/.vsvimrc)
2. Open visual studio and edit file

Via command
1. Open visual studio and edit file
2. Type `:set relativenumber` as a command

I've tried the above scenarios with the `Text Editor->Markers and Rulers->Show line number` enabled and disabled. 

**Expected behavior**
Following vim's `relativenumber` option: line numbers are shown on the left side of the editor relative to the cursor's line.

**Environment (please complete the following information):**

- Visual Studio version: Visual Studio for Mac 8.4.1
- VsVim version: 2.8.0.0
- Programming Language: C#

**Comment by funseiki** [comment]
@nosami - Any luck with this one? Not a request to start working or anything, but just checking to see if this had any updates.

**Comment by nosami** [comment]
@funseiki Sorry, I haven't even taken a look what is involved yet - I'm really busy with my day job right now and will be for another couple of weeks at least.

Having said that, I'd really like to get this done. I imagine it should be a simple port of the Windows code.

**Comment by nosami** [comment]
https://user-images.githubusercontent.com/667194/117424667-ed95e900-af19-11eb-9b6e-7c5e72d295e5.mov

soon!

**Comment by nosami** [comment]
@funseiki I created a draft PR [here](https://github.com/VsVim/VsVim/issues/2771) if you'd like to help me test it out. Thanks!

**Comment by funseiki** [comment]
> @funseiki I created a draft PR [here](https://github.com/VsVim/VsVim/pull/2892) if you'd like to help me test it out. Thanks!

Sweet, I'll take a look today or tomorrow. Is this the latest build: https://www.vsixgallery.com/extension/VsVim.Microsoft.e214908b-0458-4ae2-a583-4310f29687c3/ ?

**Comment by nosami** [comment]
> Is this the latest build:

That is the link to the vsix file that is used in Windows. 

You need the mpack file extracted from this zip file https://dev.azure.com/VsVim/503f82b4-a499-4dec-abba-84bfda309969/_apis/build/builds/562/artifacts?artifactName=VSMacExtension&api-version=6.0&%24format=zip

**Comment by funseiki** [comment]
@nosami Looks great. I'll attach an image of what I'm seeing. Both `:set nu` as well as `:set rnu` work great!

The only issue I'm seeing is with disabing/enabling `Show line numbers` under `Markers and Rulers`. Doing so doesn't have an effect. Otherwise, this is perfect.

<img width="508" alt="Screen Shot 2021-05-19 at 3 52 26 PM" src="https://user-images.githubusercontent.com/669439/118875592-39517680-b8ba-11eb-8747-051760ee6aef.png">

**Comment by nosami** [comment]
> The only issue I'm seeing is with disabing/enabling Show line numbers under Markers and Rulers. Doing so doesn't have an effect.

I was wondering about this when implementing. In real vim, `set nu!` will hide the number margin completely when both `nu` and `rnu` are unset.

I guess it should probably hide the margin when both the setting is unchecked and both `nu` and `rnu` are unset.

**Comment by funseiki** [comment]
So would the logic be:
showMargin = (settingChecked) && (nuSet || rnuSet)

**Comment by nosami** [comment]
I think the problem might be that if neither `nu` or `rnu` are set, but the user has the setting checked in preferences to display line numbers, then it might be confusing if the margin is not displayed. Not sure what is the best approach here.

**Comment by funseiki** [comment]
Ah yeah that would be confusing.

Is it possible to set nu on tick of the checkbox if neither nu nor rnu are set? It's not super elegant, but at least this way the setting acts as a global off switch and a reasonable on switch.

E.g.

```
OnSettingTicked():
   if(!nu && !rnu)
      setNu()

   DisplayMargin()
```

**Comment by nosami** [comment]
> Is it possible to set nu on tick of the checkbox if neither nu nor rnu are set?

Not really. That is handled by VSMac core code.

There are actually 2 margins. The original margin is actually hidden right now. 
Maybe the checkbox should control the original margin and the vim line number options should control the vim line number margin? This does mean that it would be possible to see both margins simultaneously.

Alternatively, the checkbox setting could maybe only be used if neither `nu` or `rnu` are used.

**Comment by funseiki** [comment]
Yeah, I think that's fine if both are simultaneously showing, though maybe others can chime in.

I am seeing a weird issue with this build, though, where I'm not getting matching curlies to appear. Matching parentheses do show up, though. I also am not seeing the formatting happen when I place a brace even though I have that checked. This issue goes away when I disable VsVim.

**Comment by nosami** [comment]
> Yeah, I think that's fine if both are simultaneously showing, though maybe others can chime in.

I wish others would chime in, I suspect that we are in a bit of a niche here though :)

> I am seeing a weird issue with this build, though, where I'm not getting matching curlies to appear. Matching parentheses do show up, though. I also am not seeing the formatting happen when I place a brace even though I have that checked. This issue goes away when I disable VsVim.

Please could you try the following build? It is the current release (2.8.0.12) with the relative numbers feature added on top.
https://dev.azure.com/VsVim/503f82b4-a499-4dec-abba-84bfda309969/_apis/build/builds/571/artifacts?artifactName=VSMacExtension&api-version=6.0&%24format=zip

**Comment by funseiki** [comment]
> I wish others would chime in, I suspect that we are in a bit of a niche here though :)

Hah, yeah that might be true! Simultaneous makes sense to me then. It can always be disabled in favor of vim only (which is what I'll be doing).

> Please could you try the following build? It is the current release (2.8.0.12) with the relative numbers feature added on top.
> https://dev.azure.com/VsVim/503f82b4-a499-4dec-abba-84bfda309969/_apis/build/builds/571/artifacts?artifactName=VSMacExtension&api-version=6.0&%24format=zip

Nice, this is fixed for me now! Thanks :)

**Comment by nosami** [comment]
> Hah, yeah that might be true! Simultaneous makes sense to me then. It can always be disabled in favor of vim only (which is what I'll be doing).

I think this is probably the best solution. At least the one that makes the most sense when I try and explain how it works ;) I'll get this done in the next couple of days and get it released.

**Comment by nosami** [comment]
This is available in version 2.8.0.13

It can take up to 24 hrs for it to become available in the VSMac update feed.
